semgrep/semgrep/profile_manager.py

```diff
@@ -0,0 +1,22 @@
+import collections
+import time
+from typing import Any
+from typing import Dict
+from typing import List
+
+
+class ProfileManager:
+    """
+    An incredibly rudimentary tool for tracking profiled calls with an associated named key
```

Comment:
I have exactly the same poor's man profiler in pfff in Common.profile_code and now also its annitation [@@profiling]

Semgrep-core -profile returns such info with stats per rule, ststs for parsing, naming, etc.